frontend/src/interface/projectInterface.js

```diff
@@ -29,44 +35,98 @@ export function getModels() {
 }
 
 /**
- * should get the material compositon of a specific model
+ * Gets the filtered Material Composititon Data from API
  * inputs contribibuting less than 1% each, should have been filtered out by Siemens Energy
- * @param modelName name of the model, which we want to get the Data
+ * Calculates the percentage values and returns them.
+ * @param compositionData filtered data from backendconnnect
  */
-export function getMaterialCompositionData(modelName) {
-    return [17, 13, 3, 2, 1, 42, 21];
+export function setMaterialCompositionData(compositionData) {
+    console.log('compositionData');
+    materialCompositionData = Array.from(compositionData);
+    console.log(materialCompositionData);
+    let sum = 0;
+    for (let i = 0; i < materialCompositionData.length; i++) {
+        sum += Number(materialCompositionData[i]);
+    }
+    console.log(sum);
+    for (let i = 0; i < materialCompositionData.length; i++) {
+        materialDataInPercent[i] = (Number(materialCompositionData[i]) / sum) * 100;
+    }
+    console.log(materialDataInPercent);
 }
+
 /**
- * @param modelName name of the model, which we want to get the Data
+ * Getter method to recieve the filtered Material Composititon Data from API
  */
-export function getMaterialCompositionLabels(modelName) {
-    return [
-        'Transformer oil',
-        'Steel',
-        'Pressboard',
-        'Stainless steel',
-        'Aluminium',
-        'Silicon steel',
-        'Copper'
-    ];
+export async function getMaterialCompositionData() {
+    return materialDataInPercent;
 }
 
 /**
- * QUESTON: Do these categories stay the same?
+ * Gets the filtered Material Composititon Labels from API
+ * Calculates the percentage values and returns them.
+ * @param compositionData filtered data from backendconnnect
+ */
+export function setMaterialCompositionLabels(compositionLabels) {
+    console.log('compositionLabels');
+
+    materialCompositionLabels = Array.from(compositionLabels);
+
+    console.log(materialCompositionLabels);
+}
+
+/**
+ * Getter method to recieve the filtered Material Composititon Labels from API
+ */
+export async function getMaterialCompositionLabels() {
+    return materialCompositionLabels;
```

Comment:
Have we implemented the color mapping for the materials yet?

Reply:
I don't think so

Reply:
This was updated yesterday and the changes were reverted. As we do not want to receive the label data from API